posix.mak

```diff
@@ -626,10 +616,47 @@ $(TESTS_EXTRACTOR): $(TOOLS_DIR)/tests_extractor.d | $(LIB)
 # This is done to check for potentially missing imports in the examples, e.g.
 # make -f posix.mak std/format.publictests
 ################################################################################
+
+publictests: $(addsuffix .publictests,$(D_MODULES))
+
 %.publictests: %.d $(LIB) $(TESTS_EXTRACTOR) | $(PUBLICTESTS_DIR)/.directory
 	@$(TESTS_EXTRACTOR) --inputdir  $< --outputdir $(PUBLICTESTS_DIR)
 	@$(DMD) $(DFLAGS) $(NODEFAULTLIB) $(LIB) -main $(UDFLAGS) -run $(PUBLICTESTS_DIR)/$(subst /,_,$<)
 
+################################################################################
+# Check and run @betterC tests
+# ----------------------------
+#
+# Extract @betterC tests of a module and run them in -betterC
+#
+#   make -f posix.mak std/format.betterc
+################################################################################
+
+betterc-phobos-tests: $(addsuffix .betterc,$(D_MODULES))
+betterc: betterc-phobos-tests betterc-run-tests
+
+%.betterc: %.d | $(BETTERCTESTS_DIR)/.directory
+	@# Due to the FORCE rule on druntime, make will always try to rebuild Phobos (even as an order-only dependency)
+	@# However, we still need to ensure that the test_extractor is built once
+	@[ -f "$(TESTS_EXTRACTOR)" ] || ${MAKE} -f posix.mak "$(TESTS_EXTRACTOR)"
+	@$(TESTS_EXTRACTOR) --betterC --attributes betterC \
+		--inputdir  $< --outputdir $(BETTERCTESTS_DIR)
+	@$(DMD) $(DFLAGS) $(NODEFAULTLIB) -betterC $(UDFLAGS) -run $(BETTERCTESTS_DIR)/$(subst /,_,$<)
+
+################################################################################
+# Run separate -betterC tests
+################################################################################
+
+test/betterC/%.run: test/betterC/%.d $(DMD) $(LIB)
+	mkdir -p $(ROOT)/unittest/betterC
+	$(DMD) $(DFLAGS) -of$(ROOT)/unittest/betterC/$(notdir $(basename $<)) -betterC $(UDFLAGS) \
+		-defaultlib= -debuglib= $(LINKDL) $<
+	./$(ROOT)/unittest/betterC/$(notdir $(basename $<))
+
+betterc-run-tests: $(subst .d,.run,$(wildcard test/betterC/*.d))
```

Comment:
There's probably not much value in keeping these separate tests anymore as `@betterC` is almost as powerful and we avoid lots of redundant declarations.

Reply:
Yep, I'm ok with removing this target.